### DISCUSSION TITLE
Github is driving me nuts

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#296] [General]
Not the fault of Github, though. Just the accessibility to Github from my area is greatly reduced. VPNs don't help. The connection speed is too slow, making cloning from Github take too much time. I discovered shallow clone with `--depth 1` and it really helped. But I can no longer make my own local branch and safely revert everything to upstream like what I can do with a full clone. The Objeck repository has grown too large, in my opinion. It seems to be because of your habit of making many small commits that the history (metadata) grows too fast. Most of the bloat comes from the metadata so a shallow clone really reduced the amount of the data needs to be fetched.

The compilation time of Objeck is also worsening...

Long story short, I can't do a local build to test the latest features anymore.

**Comment by objeck** [comment]
Thank you for the feedback. My development workstation is still down and needs to be rebuilt. It will hopefully be back up before the weekend. I will look into the size of the code repository and build times once I have a working environment. I am currently working off a laptop, however, running VMs is painfully slow and required too much memory.

**Reply by ghost** [reply]
The rule of thumb is to do backups before messing with Git. Git is so dreadful. BTW, I think you shouldn't mess with it at all. The slow cloning problem is mostly due to the restricted access to Github in my area, and I think it will not improve anytime soon.

**Comment by objeck** [comment]
Got it, thank you for the feedback!
